src/emailjs-tcp-socket.js

```diff
@@ -386,6 +386,49 @@
     }
 
     function chromeShim() {
+
+        // setZeroTimeout slightly adapted from
+        // https://github.com/shahyar/setZeroTimeout-js (CC BY 3.0).
+        // Provides a function similar to setImmediate() on Chrome.
+        var setZeroTimeout = (function (w) {
+            var timeouts = [],
+            msg_name = 'asc0tmot',
+
+            // Like setTimeout, but only takes a function argument.  There's
+            // no time argument (always zero) and no arguments (you have to
+            // use a closure).
+            _postTimeout = function (fn) {
+                timeouts.push(fn);
+                postMessage(msg_name, '*');
+            },
+
+            _handleMessage = function (event) {
+                if (event.source === w && event.data === msg_name) {
+                    if (event.stopPropagation) {
+                        event.stopPropagation();
+                    }
+                    if (timeouts.length) {
+                        try {
+                            timeouts.shift()();
+                        } catch (e) {
+                            // Throw in an asynchronous closure to prevent setZeroTimeout from hanging due to error
+                            setTimeout((function (e) {
+                                return function () {
+                                    throw e.stack || e;
+                                };
+                            }(e)), 0);
+                        }
+                    }
+                    if (timeouts.length) { // more left?
+                        postMessage(msg_name, '*');
+                    }
+                }
+            };
+
+            addEventListener('message', _handleMessage, true);
+            return _postTimeout;
+        }(window));
+
         TCPSocket = function(config) {
```

Comment:
is there any advantage of using this over setTimeout(fn, 0) other than speed? the reason for this is to queue this at the next start of the next event loop iteration, i'd think a fraction of a millisecond more or less shouldn't make any problems here, right?